src/polyswarm_api/__init__.py

```diff
@@ -192,62 +205,52 @@ def _fix_result(self, result):
+        :param artifact_data_objs: List of file-like object to POST to the API
+        :param artifact_data_names: List of names of data to be given to the API
         :param artifact_type: Type of artifact being posted
         :return: Dictionary of the result code and the UUID of the upload (if successful)
         """
         # TODO check file-size. For now, we need to handle error.
         data = aiohttp.FormData()
-        data.add_field('file', artifact_data_obj, filename=artifact_data_name)
+        printable_artifact_names = ', '.join(artifact_data_names)
+        for artifact_data_obj, artifact_data_name in zip(artifact_data_objs, artifact_data_names):
+            logger.debug('Adding file %s to request', artifact_data_name)
+            data.add_field('file', artifact_data_obj, filename=artifact_data_name)
         data.add_field('artifact-type', artifact_type.name)
 
         params = {'force': 'true'} if self.force else {}
         async with self.post_semaphore:
-            logger.debug('Posting artifact %s with api-key %s', artifact_data_name, self.api_key)
+            logger.debug('Posting artifacts %s with api-key %s', printable_artifact_names, self.api_key)
             async with aiohttp.ClientSession() as session:
+                # prepare the failure message in case it is needed
+                failure_message = 'Failed to get UUID for scan of files %s'.format(printable_artifact_names)
                 try:
                     async with session.post(self.community_uri, data=data, params=params,
                                             headers={'Authorization': self.api_key}) as raw_response:
                         try:
                             response = await raw_response.json()
-                        except Exception:
+                        except Exception as e:
                             response = await raw_response.read() if raw_response else 'None'
                             logger.error('Received non-json response from PolySwarm API: %s', response)
-                            response = {'filename': artifact_data_name, 'result': 'error'}
+                            raise exceptions.BadFormatException(failure_message) from e
+                        # check for non-200-ish responses
                         if raw_response.status // 100 != 2:
                             errors = response.get('errors')
                             logger.error('Error posting to PolySwarm API: %s', errors)
-                            response = {'filename': artifact_data_name, 'status': 'error'}
+                            raise exceptions.ServerErrorException(failure_message)
+                        # check if the server responded with anything but "OK"
+                        if response['status'] != 'OK':
+                            logger.error('Failed to get UUID for scan')
+                            raise exceptions.ServerErrorException(failure_message)
```

Comment:
I see you're handling these exceptions in the caller functions currently; I would prefer the exception handler for that moved back in here so that people calling this directly don't now need to handle these, as the result should always be the same error dict and updating all of those handlers will be annoying if we ever change the error result dict.

This change initially worried me a lot, as I think it's a rather drastic change to our API behavior to have a public method raise exceptions where it previously would handle them gracefully. However, like I mention in another comment, this particular method probably should just be private. I'd like to discuss offline whether that behavior (raising exceptions in public methods) is what we actually want in the API, as it would need to be worked into the current larger refactor.

Reply:
Will treat the exception inside the `post_artifact` method for now. As a python developer, I expect errors to throw exceptions, and not to have to check the return value of functions. I think we can discuss how to improve this in the future.